### ISSUE TITLE
[Feature]: add function_call result for none gpt* models

### ISSUE THREAD

**Issue by peterz3g** [BerriAI/litellm#664]
### The Feature

here is the function input example for none gpt* models. it is very good!
https://docs.litellm.ai/docs/completion/function_call

but
==========================================
for the 【gpt model】 response for function config input is like below, which contains key=function_call.
-------------------------------------
reply['function_call']
<OpenAIObject at 0x7f400379a900> JSON: {
  "name": "get_current_weather",
  "arguments": "{\n  \"location\": \"Boston, MA\"\n}"
}

==========================================
for 【none gpt】 model response, the result is :
----------------------------------------------
choices[0]['message']['content']
" To get the current weather in Boston, I would call the get_current_weather function like this:\n\n```python\nget_current_weather({'location': 'Boston, MA'})\n```\n\nSince the unit parameter is not specified, it will default to fahrenheit."

==========================================
I want the none gpt mode response just the same as gpt mode response. which must contains function_call field.

thanks

### Motivation, pitch

if has this feature. all none gpt models can work as gpt models to support function config. not only input, bug also output.

**Comment by ghost** [comment]
cc: @ishaan-jaff

**Comment by ishaan-jaff** [comment]
@peterz3g here's how you can do function calling for non openai llms: https://docs.litellm.ai/docs/completion/function_call#function-calling-for-non-openai-llms 

We add the function call to the prompt when you set `litellm.add_function_to_prompt = True ` 

did this not work for you?

**Comment by peterz3g** [comment]
@ishaan-jaff  No, I want the response from the non openai llms is the same format as openai llms.
================openai llm response ==================
response: {
  "id": "chatcmpl-xxx",
  "object": "chat.completion",
  "created": 1697933713,
  "model": "gpt-3.5-turbo-0613",
  "choices": [
    {
      "index": 0,
      "message": {
        "role": "assistant",
        "content": null,
        "function_call": {
          "name": "get_current_weather",
          "arguments": "{\n  \"location\": \"Chengdu\"\n}"
        }
      },
      "finish_reason": "function_call"
    }
  ],
  "usage": {
    "prompt_tokens": 83,
    "completion_tokens": 18,
    "total_tokens": 101
  }
}

================non openai llm response ==================
response: {
  "object": "chat.completion",
  "choices": [
    {
      "finish_reason": "stop_sequence",
      "index": 0,
      "message": {
        "content": " {\n  \"name\": \"get_current_weather\",\n  \"arguments\": {\n    \"location\": \"Chengdu, China\"\n  },\n  \"prompt\":\"The current weather in Chengdu: Mostly cloudy with a high of 25\u00b0C (77\u00b0F) and low of 18\u00b0C (64\u00b0F).\"\n}",
        "role": "assistant"
      }
    }
  ],
  "id": "chatcmpl-xxx",
  "created": 1697933664.5596185,
  "model": "claude-instant-1",
  "usage": {
    "prompt_tokens": 491,
    "completion_tokens": 64,
    "total_tokens": 555
  }
}
------------------------------------------
and I am writing a wrapper to process the function call response now, like this:
1. I modify the factory.py/function_call_prompt code, to format the llm response
2. I add a wrapper befor main.py/completion, to process all the llm response result, so that the result is the same as gpt.

      "message": {
        "content": null,
        "role": "assistant",
        "function_call": {
          "name": "get_current_weather",
          "arguments": "{\"location\": \"Chengdu, China\", \"unit\": \"celsius\"}"
        }
  "model": "claude-instant-1" 's responce message is the same as gpt now.

**Comment by ghost** [comment]
@peterz3g how do we reliably do this parsing?

What happens if a model doesn't reply with a valid function response?

**Comment by peterz3g** [comment]
> @peterz3g how do we reliably do this parsing?
> 
> What happens if a model doesn't reply with a valid function response?

I have compared the llm response between gpt and other llm. And do some task to  ensure stability
1. write prompt to make sure the llm response is some determined format
2. in the response parsing wrapper, handle different situations

**Comment by ghost** [comment]
@peterz3g not entirely sure what that means. open to a PR here, if you'd like to contribute!

**Comment by peterz3g** [comment]
> @peterz3g not entirely sure what that means. open to a PR here, if you'd like to contribute!

OK, I'd be happy to contribute. I'll take into account some of the things I found in my tests.

**Comment by ghost** [comment]
thanks!

**Comment by peterz3g** [comment]
@krrishdholakia 
the commit is here：https://github.com/BerriAI/litellm/pull/768
